### PR TITLE
[query] Support dynamic namespace resolution for embedded coordinators

### PR DESCRIPTION
**Pull request by nbroyles**
**What this PR does / why we need it**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing and/or backwards incompatible change?**:

```release-note

```

**Does this PR require updating code package or user-facing documentation?**:

```documentation-note

```

### CODE REVIEW

File: src/query/server/query.go [hunk 1]
```diff
@@ -833,13 +826,19 @@ func initClusters(
 		clustersCfg := m3.ClustersStaticConfiguration{
 			m3.ClusterStaticConfiguration{
 				Namespaces: localCfg.Namespaces,
+				Client:     client.Configuration{EnvironmentConfig: &dbCfg.EnvironmentConfig},
```
Comment (arnikola) [inline src/query/server/query.go:829]: Think this may need a nil check

**Review by arnikola — APPROVED** [review]
LGTM with a nil check

File: src/query/server/query.go [hunk 2]
```diff
@@ -830,16 +823,27 @@ func initClusters(
...
+		cfgOptions := m3.ClustersStaticConfigurationOptions{
 			ProvidedSession:    session,
 			CustomAdminOptions: customAdminOptions,
-		}, clusterNamespacesWatcher)
+		}
+
+		if staticNamespaceConfig {
+			clusters, err = clustersCfg.NewStaticClusters(instrumentOpts, cfgOptions, clusterNamespacesWatcher)
+		} else {
+			clusters, err = clustersCfg.NewDynamicClusters(instrumentOpts, cfgOptions, clusterNamespacesWatcher)
```
Comment (wesleyk) [inline src/query/server/query.go:845]: (reading along to understand, excuse the naive-ty!) it's not clear to me why a `ClustersStaticConfiguration` would be responsible for exposing `NewDynamicClusters` if it's responsible for static config

Comment (nbroyles) [reply src/query/server/query.go:845]: Heh, this is a good call out. `ClustersStaticConfiguration` should probably just be renamed `ClustersConfiguration` as the static in its name merely points out that this is a static list of clusters which is what anything is that's defined in a config file, so it's a bit redundant.

The purpose of `ClustersStaticConfiguration` is to generate a Clusters instance which is an object that can be used to get the namespaces in a cluster. A Clusters instance can be static (i.e. namespaces are defined in config) vs dynamic (i.e. namespaces are fetched from etcd). Hence the `NewStaticClusters` and `NewDynamicClusters` instance.

I should be tinkering in here over the next week or two. Will refactor and send a PR your way.

**Pull request merged**